src/topology/topological_fiber_bundle.lean

```diff
@@ -979,3 +1049,91 @@ begin
 end
 
 end topological_fiber_bundle_core
+
+section topological_fiber_prebundle
+
+variables (F) {Z : Type*} [topological_space B] [topological_space F] {proj : Z → B}
+
+/-- This structure permits to define a fiber bundle when trivializations are given as local
+equivalences but there is not yet a topology on the total space. The total space is hence given a
+topology in such a way that there is a fiber bundle structure for which the local equivalences
+are also local homeomorphism and hence local trivializations. -/
+@[nolint has_inhabited_instance]
+structure topological_fiber_prebundle (proj : Z → B) :=
+(trivialization_at : B → prebundle_trivialization F proj)
+(mem_base_trivialization_at : ∀ x : B, x ∈ (trivialization_at x).base_set)
+(continuous_triv_change : ∀ x y : B, continuous_on ((trivialization_at x) ∘
+  (trivialization_at y).to_local_equiv.symm) ((trivialization_at y).target ∩
+  ((trivialization_at y).to_local_equiv.symm ⁻¹' (trivialization_at x).source)))
+
+variable {F}
+
+/-- Topology on the total space that will make the prebundle into a bundle. -/
+def topological_fiber_prebundle.total_space_topology (a : topological_fiber_prebundle F proj) :
+  topological_space Z :=
+⨆ x : B, coinduced (a.trivialization_at x).set_symm (subtype.topological_space)
+
+lemma topological_fiber_prebundle.continuous_inv_triv_at (a : topological_fiber_prebundle F proj)
```

Comment:
instead of `inv_triv_at`, can you use `symm_trivialization_at`? (The name should be as guessable as possible, here it is referring to the `symm` of `trivialization_at`). Same thing in next lemmas. Also, the order of words in the lemma name should correspond to the order of words in the statement, again for improved guessability (is that a word ? :-) So `triv_at_open_inter` should probably be `is_open_target_trivialization_at_inter` or something like that. (don't try to maximize usability for you author when you write a code, but think first of the user!)